### PR TITLE
[ISSUE #147]An enhancement about the convert in RocketMQTemplate

### PR DESCRIPTION
**Pull request by zkzlx**
## What is the purpose of the change

An enhancement about the convert in RocketMQTemplate

## Brief changelog

enhancement  #147 

Do not use `ObjectMapper` directly, Instead, the externally built converter is injected into `MessageConverter` in  `RocketMQTemplate`.

a piece of code in `RocketMQMessageConverter` :
```java
public RocketMQMessageConverter() {
	List<MessageConverter> messageConverters = new ArrayList<>();
	ByteArrayMessageConverter byteArrayMessageConverter = new ByteArrayMessageConverter();
	byteArrayMessageConverter.setContentTypeResolver(null);
	messageConverters.add(byteArrayMessageConverter);
	messageConverters.add(new StringMessageConverter());
	if (JACKSON_PRESENT) {
		messageConverters.add(new MappingJackson2MessageConverter());
	}
	if (FASTJSON_PRESENT) {
		try {
			messageConverters.add(
				(MessageConverter)ClassUtils.forName(
					"com.alibaba.fastjson.support.spring.messaging.MappingFastJsonMessageConverter",
					ClassUtils.getDefaultClassLoader()).newInstance());
		} catch (ClassNotFoundException | IllegalAccessException | InstantiationException e) {
		}
	}
	messageConverter = new CompositeMessageConverter(messageConverters);
}
```

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. 
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by zkzlx** [comment]
@vongosling Please take a look

File: rocketmq-spring-boot-samples/rocketmq-produce-demo/src/main/java/org/apache/rocketmq/samples/springboot/ProducerApplication.java [hunk 1]
```diff
@@ -69,8 +74,15 @@ public void run(String... args) throws Exception {
         SendResult sendResult = rocketMQTemplate.syncSend(springTopic, "Hello, World!");
         System.out.printf("syncSend1 to topic %s sendResult=%s %n", springTopic, sendResult);
 
+        SendResult sendResult2 = rocketMQTemplate.syncSend(userTopic, new User().setUserAge((byte)18).setUserName("Kitty"));
+        System.out.printf("syncSend1 to topic %s sendResult=%s %n", userTopic, sendResult2);
+
+        SendResult sendResult3 = rocketMQTemplate.syncSend(userTopic, MessageBuilder.withPayload(
+            new User().setUserAge((byte)21).setUserName("Lester")).setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.APPLICATION_JSON_VALUE).build());
+        System.out.printf("syncSend1 to topic %s sendResult=%s %n", userTopic, sendResult3);
+
```
Comment (RongtongJin) [inline rocketmq-spring-boot-samples/rocketmq-produce-demo/src/main/java/org/apache/rocketmq/samples/springboot/ProducerApplication.java]: sendResult variables can be reused.

File: rocketmq-spring-boot-samples/rocketmq-consume-demo/src/main/java/org/apache/rocketmq/samples/springboot/consumer/User.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "User{" +
+            "userName='" + userName + '\'' +
+            ", userAge=" + userAge +
+            '}';
+    }
+}
```
[the remark this answers is not in the document]
Comment (zkzlx) [reply rocketmq-spring-boot-samples/rocketmq-consume-demo/src/main/java/org/apache/rocketmq/samples/springboot/consumer/User.java]: Sorry, I don't understand what **domain package** means here .

[the remark this answers is not in the document]
Comment (RongtongJin) [reply rocketmq-spring-boot-samples/rocketmq-consume-demo/src/main/java/org/apache/rocketmq/samples/springboot/consumer/User.java]: > Sorry, I don't understand what **domain package** means here .

org.apache.rocketmq.samples.springboot.domain, OrderPaidEvent.java  is placed there.

[the remark this answers is not in the document]
Comment (zkzlx) [reply rocketmq-spring-boot-samples/rocketmq-consume-demo/src/main/java/org/apache/rocketmq/samples/springboot/consumer/User.java]: > > Sorry, I don't understand what **domain package** means here .
> 
> org.apache.rocketmq.samples.springboot.domain, OrderPaidEvent.java is placed there.

OK,Thanks. 
Are there any other problems besides those that have been found? I will finish them tonight .

**Comment by zkzlx** [comment]
@RongtongJin @vongosling I was solved , you can check it now. thanks .

**Comment by RongtongJin** [comment]
LGTM~

**Comment by vongosling** [comment]
@zkzlx Could you help to resolve the conflict :-)

**Pull request merged**